homeassistant/components/growatt_server/sensor.py

```diff
@@ -0,0 +1,123 @@
+"""Read status of growatt inverters."""
+import datetime
+import logging
+import growattServer
+import voluptuous as vol
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_USERNAME = "username"
+CONF_PASSWORD = "password"
+CONF_PLANT_ID = "plant_id"
+DEFAULT_UNIT = "kW"
+DEFAULT_NAME = "Growatt"
+SCAN_INTERVAL = datetime.timedelta(minutes=5)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_PLANT_ID, default="0"): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Growatt sensor."""
+    username = config.get(CONF_USERNAME)
```

Comment:
Please use `dict[key]` for required config keys and keys with default config schema values.